src/components/Home/Home.js

```diff
@@ -3,23 +3,23 @@ import PropTypes from 'prop-types';
 import { Grid } from '@material-ui/core';
 import { useEffect, useState } from 'react';
 
-const Home = ({ initTikToks, removeTikTok }) => {
-  const convertToPinObjects = initTikToks.map(tiktok => {
+const Home = ({ fetchedTTS, removeTikTok }) => {
+  const convertToPinObjects = fetchedTTS.map(tiktok => {
     return {
-      url: tiktok,
+      ...tiktok,
       isPinned: false
     }
   })
   
   const [order, setOrder] = useState([...convertToPinObjects]);
 
-  const addPin = (url) => {
-    order[order.findIndex((tiktok) => tiktok.url === url)].isPinned = true;
+  const addPin = (id) => {
+    order[order.findIndex((tiktok) => tiktok.data_video_id === id)].isPinned = true;
     sortIfPinned();
   }
 
-  const removePin = (url) => {
-    order[order.findIndex((tiktok) => tiktok.url === url)].isPinned = false;
+  const removePin = (id) => {
+    order[order.findIndex((tiktok) => tiktok.data_video_id === id)].isPinned = false;
```

Comment:
nice chaining on these two methods.

Reply:
I feel like they could be condensed into a toggle. What do you think? Maybe one function that looks something like:
```!order[order.findIndex((tiktok) => tiktok.data_video_id === id)].isPinned
sortIfPinned()
```

Reply:
I've been trying to stay away from if/else conditionals which as you've seen results in some clear but somewhat WET code, I've since refactored it to this ternary on the Independent component:
<code>
const togglePin = () => {
    isPinned ? removePin(data_video_id) : addPin(data_video_id)
  }
  </code>
 But if I am following you correctly we can toggle the re-assignment with the bang operator? Thoughts on something like this?:
<code>
const togglePin = (id) => {
  const desiredIndex = order.findIndex((tiktok) => tiktok.data_video_id === id);
  order[desiredIndex].isPinned = !order[desiredIndex].isPinned
}
</code>

Reply:
Yep, that's right!  🔥